src/util/xml.h

```diff
@@ -27,6 +27,13 @@ class xmlt
   typedef std::list<xmlt> elementst;
   typedef std::map<std::string, std::string> attributest;
 
+  xmlt(std::string _name, attributest &&_attributes, elementst &&_elements)
+    : name(std::move(_name)),
```

Comment:
Should _name get a `&&`?

Reply:
I was wondering how well that would work given the common case is a C string at that point. Or maybe it actually makes no difference?

Reply:
std::string has a lot of optimisation behind it (I've seen instances with copy-on-write, short-string stored in the class itself, etc.); however, disregarding this, yes, it should.